keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/ConcatenationVertex.java

```diff
@@ -139,7 +139,7 @@ private static DoubleTensor concatPartialDerivatives(int dimension, List<DoubleT
             splitPartials.put(operands[i], new PartialDerivatives(new HashMap<>()));
         }
 
-        int wrtDimensionToSliceOn = operands[0].getShape().length + dimension;
+        int wrtDimensionToSliceOn = -operands[0].getShape().length + dimension;
```

Comment:
It's not immediately obvious how this works - comment or one extra step?

Reply:
👍